### DISCUSSION TITLE
Any good way to bundle the web part?

### DISCUSSION THREAD

**Discussion by OsakiTsukiko** [webui-dev/webui#490] [Q&A]
Is there any "official" or good way to bundle the whole project in something like an archive (like electron does) or even inside the binary itself? I do know about the file handler function on webui but building my own bundler seems like a lot of work.

**Comment by wwderw** [comment]
I tend to embed in the binary itself.  Some languages (Nim, Zig, Go (there may be others, those are just the languages that I have experience with) have builtin ways to do that and use the handler function to display them.  

The most cross platform way for C/C++ is to create a char array in a header file (downside to this is that the it adds significantly to the binary, for instance what is an originally a 200kb picture can end up being a 5MB in the program).  Could also use ld from GCC to link them as object files and embed them that way.  I've even just used base64 in variables to store them (still a hit size wise, but nothing like header approach).

I have not really done the archive approach.  As long as the final binary isn't astronomical in size, I prefer the embed in binary approach myself, but that could just be me.

**Comment by OsakiTsukiko** [comment]
I'm mainly using zig which has `@embedFile`, but there are many files in my project and embeding them all manually would be a lot of work, also there is the issue with non standard files, like svgs and such, ig sending the raw bytes would work but still manually embeding each file would be a pain. Zig has comp time so that could maybe work but im not sure.

**Reply by wwderw** [reply]
Go does allow embedding folders for a virtual file system as well.  Zig I have only done each file.  It is tedious, but given what I am used to with C/C++ no big deal.

**Reply by OsakiTsukiko** [reply]
ig best option is writing an [asar](https://github.com/electron/asar) wrapper

**Comment by AlbertShown** [comment]
In C, you can use this example:
https://github.com/webui-dev/webui/tree/main/examples/C/virtual_file_system

It has a simple Python script that automatically create a C header file of all the project's files.
